File: example/Tensor/clone.py
import sys
from pathlib import Path
home = str(Path.home())
sys.path.append(home + '/Cytnx_lib')
from cytnx import *

A = Tensor([3,4,5])

B = A
C = A.clone()

print(B is A)
print(C is A)

File: example/Tensor/clone.cpp
#include "cytnx.hpp"
#include <iostream>

using namespace cytnx;
int main() {
  Tensor A({3, 4, 5});

  Tensor B = A;  // B shares same object with A
  Tensor C = A.clone();  // C is a copy of A

  // use is() to check if two variable shares same object
  std::cout << is(B, A) << std::endl;
  std::cout << is(C, A) << std::endl;

  return 0;
}
